Makefile

```diff
@@ -118,398 +132,424 @@ E2E_CONTENT_IMAGE_PATH?=ghcr.io/complianceascode/k8scontent:latest
+.PHONY: help
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
 .PHONY: clean
-clean: clean-modcache clean-cache clean-output ## Clean the golang environment
+clean: clean-modcache clean-cache clean-output clean-test clean-kustomize ## Run all of the clean targets.
 
 .PHONY: clean-output
-clean-output:
-	rm -rf $(TARGET_DIR)
+clean-output: ## Remove the operator bin.
+	rm -f $(TARGET_OPERATOR)
 
 .PHONY: clean-cache
-clean-cache:
+clean-cache: ## Run go clean -cache -testcache.
 	$(GO) clean -cache -testcache $(PKGS)
 
 .PHONY: clean-modcache
-clean-modcache:
+clean-modcache: ## Run go clean -modcache.
 	$(GO) clean -modcache $(PKGS)
 
-.PHONY: fmt
-fmt:  ## Run the `go fmt` tool
-	@$(GO) fmt $(PKGS)
+.PHONY: clean-test
+clean-test: clean-cache ## Clean up test cache and test setup artifacts.
+	rm -rf $(TEST_SETUP_DIR)
+
+.PHONY: clean-kustomize
+clean-kustomize: ## Reset kustomize changes in the repo.
+	@git restore bundle/manifests/compliance-operator.clusterserviceversion.yaml config/manager/kustomization.yaml
 
 .PHONY: simplify
-simplify:
+simplify: ## Run go fmt -s against code.
 	@gofmt -s -l -w $(SRC)
 
-.PHONY: verify
-verify: vet mod-verify gosec ## Run code lint checks
+fmt: ## Run go fmt against code.
+	$(GO) fmt ./...
 
-.PHONY: vet
-vet:
-	@$(GO) vet $(PKGS)
+vet: ## Run go vet against code.
+	$(GO) vet ./...
 
-.PHONY: mod-verify
-mod-verify:
-	@$(GO) mod verify
+.PHONY: verify
+verify: vet gosec ## Run vet and gosec checks.
 
 .PHONY: gosec
-gosec:
-	@$(GO) run github.com/securego/gosec/v2/cmd/gosec -severity medium -confidence medium -quiet ./...
+gosec: ## Run gosec against code.
+	@$(GO) run github.com/securego/gosec/v2/cmd/gosec -severity medium -confidence medium -quiet $(PKGS)
+
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+.PHONY: controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0)
+
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+.PHONY: kustomize
+kustomize: ## Download kustomize locally if necessary.
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+
+ENVTEST = $(shell pwd)/bin/setup-envtest
+.PHONY: envtest
+envtest: ## Download envtest-setup locally if necessary.
+	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
+
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
```

Comment:
Attempting to debug this locally since it doesn't appear to put controller-gen in the right place:

```
bash: line 1: /home/lbragstad/go/src/github.com/rhmdnd/compliance-operator/bin/controller-gen: No such file or directory
make: *** [Makefile:365: manifests] Error 127
```